tests/lib/acpi/src/main.c

```diff
@@ -17,14 +17,39 @@ ZTEST(acpi, test_mcfg_table)
 	zassert_not_null(mcfg, "Failed to get MCFG table");
 }
 
-ZTEST(acpi, test_irq_routing_table)
+ZTEST(acpi, test_dev_enum)
 {
-	static ACPI_PCI_ROUTING_TABLE irq_prt_table[CONFIG_ACPI_MAX_PRT_ENTRY];
-	int status;
+	struct acpi_dev *dev;
+	ACPI_RESOURCE *res_lst;
+	int ret;
 
-	status = acpi_get_irq_routing_table(CONFIG_ACPI_PRT_BUS_NAME,
-					    irq_prt_table, ARRAY_SIZE(irq_prt_table));
-	zassert_ok(status, "Failed to get PRT");
+	dev = acpi_device_get("PNP0A08", "0");
```

Comment:
These strings shouldn't be hardcoded here. *If* they're provided as part of DTS (like you're right now doing) then you should get them from DTS. If they don't come from DTS, then you should be using defines for them (but that would mean removing this stuff from DTS too).

Since this is not PCIe driver code, you can't use the DT_INST APIs. There are two other options: if we assume all platforms use the `pcie0` node label for their PCIe bus node, then you can use `DT_NODELABEL(pcie0)` instead of `DT_DRV_INST(0)`. However, if we don't want to make this assumption, you could e.g. require DTS overlays for each board in this app, and provide DT aliases in the overlay. After that you could use `DT_ALIAS()` to get the PCIe bus node (if you want to see examples, just do something like `git grep aliases tests/`).

Reply:
updated with test case enable with conditional build.